### ISSUE TITLE
Null reference error in MvxFormsAppCompatActivity on GetAccentColor

### ISSUE THREAD

**Issue by selaromdotnet** [MvvmCross/MvvmCross#2117]
I am crossposting this from SO (https://stackoverflow.com/questions/45686743/mvvmcross-and-xamarin-forms-android-app-crashes-with-null-reference-error-when) as I'm not sure if the problem I have is coming from XF, Mmx, Android, or just something wrong with my project.

I created a blank Xamarin Forms app using the Visual Studio template and added a toolbar item with an image icon. It runs fine. I updated all nuget to latest, and added MvvmCross (but didn't register the app with Mvx yet) and it still runs fine.

I then changed the app to use the Mvx Forms setup and initialization, and suddenly i'm getting a null reference error when the Xamarin.Forms library tries to call GetAccentColor. 

Like I said I don't know which change caused the error, but I did create and upload a sample project that shows the issue here: https://github.com/selaromdotnet/XamFormsAppTest

The commit history shows the changes I described above, first a blank project, then updating, then finally changing it to use Mvx.

I also found this post (https://blog.xamarin.com/material-design-for-your-xamarin-forms-android-apps/) on how to setup Material design with Xamarin Forms, which I thought might fix the problem, but I still get the same error. To be clear, I got this error immediately upon changing the app to use the Mvx stuff, before I added all the material design changes, but the error is the same.

However, the same app works fine on Windows UWP, so I'm guessing I just have something wrong setup on android, which I don't know enough about to figure out on myown.

### Steps to reproduce

1. Create a blank Xamarin Forms project using visual studio template.

2. Update nuget packages to latest, add MvvmCross

3. Update project to use MvxFormsApplication, Setup and MvxFormsAppCompatActivity

### Expected behavior
App should launch and show the same app from step 1

### Actual behavior
App crashes with null reference error:

> 08-15 01:58:19.234 I/MonoDroid( 2245): System.NullReferenceException: Object reference not set to an instance of an object.
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at Xamarin.Forms.Forms.GetAccentColor () [0x00006] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Forms.cs:159 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at Xamarin.Forms.Forms.SetupInit (Android.Content.Context activity, System.Reflection.Assembly resourceAssembly) [0x0000c] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Forms.cs:118 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at Xamarin.Forms.Forms.Init (Android.Content.Context activity, Android.OS.Bundle bundle) [0x00006] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Forms.cs:72 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Forms.Droid.MvxFormsAndroidSetup.CreateViewPresenter () [0x00000] in C:\projects\mvvmcross\MvvmCross-Forms\MvvmCross.Forms.Droid\MvxFormsAndroidSetup.cs:55 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Droid.Platform.MvxAndroidSetup.CreateViewDispatcher () [0x00000] in C:\projects\mvvmcross\MvvmCross\Droid\Droid\Platform\MvxAndroidSetup.cs:124 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Core.Platform.MvxSetup.InitializeViewDispatcher () [0x00000] in C:\projects\mvvmcross\MvvmCross\Core\Core\Platform\MvxSetup.cs:260 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Core.Platform.MvxSetup.InitializeSecondary () [0x000f8] in C:\projects\mvvmcross\MvvmCross\Core\Core\Platform\MvxSetup.cs:91 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Core.Platform.MvxSetup.Initialize () [0x00006] in C:\projects\mvvmcross\MvvmCross\Core\Core\Platform\MvxSetup.cs:37 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Droid.Platform.MvxAndroidSetupSingleton.EnsureInitialized () [0x0004e] in C:\projects\mvvmcross\MvvmCross\Droid\Droid\Platform\MvxAndroidSetupSingleton.cs:45 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at MvvmCross.Forms.Droid.MvxFormsAppCompatActivity.OnCreate (Android.OS.Bundle bundle) [0x00013] in C:\projects\mvvmcross\MvvmCross-Forms\MvvmCross.Forms.Droid\MvxFormsAppCompatActivity.cs:64 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at XamFormsAppTest.Droid.MainActivity.OnCreate (Android.OS.Bundle bundle) [0x00017] in D:\Visual Studio\OSS\XamFormsAppTest\XamFormsAppTest\XamFormsAppTest.Android\MainActivity.cs:21 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at Android.Support.V4.App.FragmentActivity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x0000f] in <8b5e0f4c6f594871ab0afcbcb2317289>:0 
> 08-15 01:58:19.234 I/MonoDroid( 2245):   at (wrapper dynamic-method) System.Object:070f481d-1d73-4674-b63e-a9c2c2385a8a (intptr,intptr,intptr)

### Configuration

**Version:** 5.1.1

**Platform:** 
- [ ] iOS
- [X] Android
- [ ] WPF
- [ ] UWP

**Comment by jamesatgithub** [comment]
@selaromdotnet what attribute do you have on the activity, I got same error and after changed the attribute like below, crash fixed. the MainTheme is defined in styles.xml

[Activity(Label = "MyAppName", Icon = "@drawable/Icon", Theme = "@style/MainTheme", ScreenOrientation = ScreenOrientation.Portrait)]

**Comment by selaromdotnet** [comment]
thanks for chiming in, i appreciate it!

so, this is what I have:

`    [Activity(Label = "XamFormsAppTest", Icon = "@drawable/icon", Theme = "@style/MainTheme", MainLauncher = true, ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation)]


I too have MainTheme defined in styles.xml and an icon.png in the drawable folder...

I tried changing the configuration changes to match yours, but still get the same error

**Comment by jamesatgithub** [comment]
@selaromdotnet forgot what I have before when got same crash but above attribute did fix it

**Comment by ryanlpoconnell** [comment]
I have a very similar (but slightly different) issue when changing the inheritance of my Forms Droid Activity to the MvxFormsAppCompatActivity.  I get the following error:

`08-18 17:09:02.784 I/MonoDroid(29608): UNHANDLED EXCEPTION:
08-18 17:09:02.806 I/MonoDroid(29608): System.NullReferenceException: Object reference not set to an instance of an object.
08-18 17:09:02.806 I/MonoDroid(29608):   at Xamarin.Forms.Platform.Android.ResourceManager.GetId (System.Type type, System.String propertyName) [0x0000d] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\ResourceManager.cs:76 
08-18 17:09:02.806 I/MonoDroid(29608):   at Xamarin.Forms.Platform.Android.ResourceManager.IdFromTitle (System.String title, System.Type type) [0x00007] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\ResourceManager.cs:70 
08-18 17:09:02.806 I/MonoDroid(29608):   at Xamarin.Forms.Platform.Android.ResourceManager.GetBitmapAsync (Android.Content.Res.Resources resource, System.String name) [0x00000] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\ResourceManager.cs:37 
08-18 17:09:02.807 I/MonoDroid(29608):   at Xamarin.Forms.Platform.Android.FileImageSourceHandler+<LoadImageAsync>d__4.MoveNext () [0x000c6] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Renderers\FileImageSourceHandler.cs:23 
08-18 17:09:02.807 I/MonoDroid(29608): --- End of stack trace from previous location where exception was thrown ---
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:151 
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00037] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.CompilerServices.TaskAwaiter1[TResult].GetResult () [0x00000] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:357 
08-18 17:09:02.807 I/MonoDroid(29608):   at Xamarin.Forms.Platform.Android.ImageRenderer+<UpdateBitmap>d__9.MoveNext () [0x000fe] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.Android\Renderers\ImageRenderer.cs:91 
08-18 17:09:02.807 I/MonoDroid(29608): --- End of stack trace from previous location where exception was thrown ---
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:151 
08-18 17:09:02.807 I/MonoDroid(29608):   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/xamarin-android/xamarin-android/external/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1018 
08-18 17:09:02.807 I/MonoDroid(29608):   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in /Users/builder/data/lanes/4695/9dbc4c53/source/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:35 
08-18 17:09:02.807 I/MonoDroid(29608):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in /Users/builder/data/lanes/4695/9dbc4c53/source/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:36 
08-18 17:09:02.807 I/MonoDroid(29608):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in /Users/builder/data/lanes/4695/9dbc4c53/source/xamarin-android/src/Mono.Android/obj/Release/android-24/mcw/Java.Lang.IRunnable.cs:81 
08-18 17:09:02.807 I/MonoDroid(29608):   at (wrapper dynamic-method) System.Object:742f4f53-f5ce-4a3a-b8c7-ed6432f0e5ea (intptr,intptr)`

If I inherit from FormsAppCompatActivity it works fine (with the additional OnCreate steps that are required in this case).

What's more interesting is if I copy the MvxFormsAppCompatActivity code file in from github and inherit from that instead, it also works without error.  I'm using MvvMCross 5.1.1 libraries, so I took that version of code from here:  https://github.com/MvvmCross/MvvmCross/blob/5.1.1/MvvmCross-Forms/MvvmCross.Forms.Droid/MvxFormsAppCompatActivity.cs . But as I say, this works!?... I guess the code version of this file is different to the one in my nuget library.

For more information, if I remove my very standard Image control from my first page the error is avoided.
`<Image Source="swoosh.png" WidthRequest="19" HeightRequest="50" HorizontalOptions="Center" VerticalOptions="Center"  />`

**Comment by mrvsoft** [comment]
Error in class MvxFormsAppCompatActivity in the OnCreate method. The method call Xamarin.Forms.Forms.Init is not from assembly are which contains resources and therefore the error occurs. (see line Assembly callingAssembly = Assembly.GetCallingAssembly(); from Xamarin.Forms.Forms.Init)

**Comment by selaromdotnet** [comment]
wow thank you for sharing this, that makes total sense, so am I correct that what you are saying is this is a bug in MvvmCross, and that Init should be called manually from the android project instead? that would explain why copying the code out of McxFormsAppCompatActivity worked for @ryanlpoconnell 

in fact, this entire rabbit hole started because in my original project I actually got the same error as above. when I made the sample project to try and duplicate the error was different...

but according to the reply I got on this post: https://stackoverflow.com/questions/45686743/mvvmcross-and-xamarin-forms-android-app-crashes-with-null-reference-error-when/45798604#45798604

the reason for THAT error is that apparently on MvvmCross Forms Android, the MvxSplashScreenActivity is REQUIRED as it does additional initialization.

I couldn't find any documentation on either of these, which has me wondering how anyone in the world has ever written a xamarin forms app with mvvmcross, unless none of them are running the latest version?

anyway, I'll try this on my original project and report back, and then try and get this working on the sample project I posted and check that in too.

thanks everyone for your help!
